### PR TITLE
Get recipe resources

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR adds | fixes:

- Move subs: get_recipe_resources and get_recipe_attributes and refactor them and their corresponding tests
- Create new sub to parse recipe prerequisites: parse_recipe_prerequisites in Recipe.pm
- Remove get_recipe_attributes and get_recipe_resources subs from recipes and use parse_recipe_prerequisites instead
- Use hash from parse_recipe_prerequisites to access recipe data in recipes

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Recipe.pm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        active_parameter_href => {
+            default     => {},
+            defined     => 1,
+            required    => 1,
+            store       => \$active_parameter_href,
+            strict_type => 1,
+        },
+        parameter_href => {
+            default     => {},
+            defined     => 1,
```
Comment (jemten) [inline lib/MIP/Recipe.pm:50]: Is defined needed here? Looks like you can use it without $parameter_href or at least an empty one

File: lib/MIP/Recipe.pm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            store       => \$parameter_href,
+            strict_type => 1,
```
Comment (jemten) [inline lib/MIP/Recipe.pm:52]: order ab

File: lib/MIP/Recipe.pm [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        },
+        recipe_name => {
+            defined     => 1,
+            required    => 1,
+            strict_type => 1,
+            store       => \$recipe_name,
```
Comment (jemten) [inline lib/MIP/Recipe.pm:58]: order ab

**Review by jemten — APPROVED** [review]
Very nice addition. I like how it simplifies the recipes 😄

File: lib/MIP/Active_parameter.pm [hunk 4]
```diff
@@ -1055,6 +1056,87 @@ sub get_package_env_cmds {
...
+    my %recipe_resource = (
+        core_number  => $core_number,
+        gpu_number   => $active_parameter_href->{recipe_gpu_number}{$recipe_name},
+        load_env_ref => \@environment_cmds,
+        memory       => $memory,
+        mode         => $active_parameter_href->{$recipe_name},
+        time         => $active_parameter_href->{recipe_time}{$recipe_name},
+    );
+
+    return $recipe_resource{$resource} if ( defined $resource && $resource );
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply lib/MIP/Active_parameter.pm:1134]: You are right and also `undef` is not allowed as a value so the process would not even get here. I will remove it

File: lib/MIP/Recipe.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipe.pm:50]: It is not required so it is ok to not supply it. However, if you do supply it it has to be defined. I think it is ok to leave it as it is.

File: lib/MIP/Recipe.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipe.pm]: fixed

**Pull request merged**